attractions/dropdown/dropdown.svelte

```diff
@@ -13,9 +22,78 @@
    * @type {boolean}
    */
   export let top = false;
+  /**
+   * Dropdown horizontal alignment relative to the shell.
+   * @type {string}
+   */
+  export let horizontalAlignment = HorizontalAlignment.AUTO_START;
+  /**
+   * Dropdown vertical alignment relative to the shell.
+   * @type {string}
+   */
+  export let verticalAlignment = VerticalAlignment.AUTO_BOTTOM;
+
+  const isDropdownOpen = getContext(CONTEXT_IS_DROPDOWN_OPEN);
+  const getDropdownShellBoundary = getContext(
+    CONTEXT_GET_DROPDOWN_SHELL_BOUNDARY
+  );
+
+  let dropdownElement, isVerticalAlignTop, isHorizontalAlignEnd;
+
+  function getIsVerticalAlignTop(dropdownBound, dropdownShellBound) {
+    const { height } = dropdownBound;
+    const { top, bottom } = dropdownShellBound;
+
+    switch (verticalAlignment) {
+      case VerticalAlignment.TOP:
+        return true;
+      case VerticalAlignment.BOTTOM:
+        return false;
+      case VerticalAlignment.AUTO_TOP:
+        return height <= top;
+      default:
+        // auto-bottom by default
+        return height > window.innerHeight - bottom;
+    }
+  }
+
+  function getIsHorizontalAlignEnd(dropdownBound, dropdownShellBound) {
+    const { width } = dropdownBound;
+    const { left } = dropdownShellBound;
+
+    switch (horizontalAlignment) {
+      case HorizontalAlignment.END:
+        return true;
+      case HorizontalAlignment.START:
+        return false;
+      case HorizontalAlignment.AUTO_END:
+        return width <= left;
```

Comment:
Then this should be `right`
```suggestion
        return width <= right;
```